framework/include/utils/MooseTypes.h

```diff
@@ -121,6 +121,14 @@ template <typename>
 class ADMaterialProperty;
 class InputParameters;
 
+enum class MaterialPropState
```

Comment:
Not suggesting a change, but it does seem to me that this is not referring to a state, but to a time.

Reply:
Not a specific time, and we call these materials "stateful". The old enum was called `Prop_State`